econml/drlearner.py

```diff
@@ -0,0 +1,633 @@
+    >>> est.effect(X[:2], T0=0, T1=1)
+    array([0.5215622 , 0.37704938])
+    >>> est.score_
+    10.243375492811202
+    >>> est.score(y, T, X=X)
+    8.489141208026698
+    >>> est.model_cate(T=1).coef_
+    array([1.00761575, 0.47127132, 0.01092897, 0.05185222])
+    >>> est.model_cate(T=2).coef_
+    array([ 1.92481336,  1.09654124,  0.08919048, -0.00413531])
+    >>> est.cate_feature_names
+    ['1', 'x0', 'x1', 'x2']
+    >>> [mdl.coef_ for mdl in est.models_regression]
+    [array([ 1.43608627e+00,  9.16715532e-04, -7.66401138e-03,  6.73985763e-01,
+             1.98864974e+00]),
+     array([ 1.49529047e+00, -2.43886553e-03,  1.74824661e-03,  6.81810603e-01,
+             2.03340844e+00])]
+    >>> [mdl.coef_ for mdl in est.models_propensity]
+    [array([[-1.05971312,  0.09307097,  0.11409781],
+            [ 0.09002839,  0.03464788, -0.09079638],
+            [ 0.96968473, -0.12771885, -0.02330143]]),
+     array([[-0.98251905,  0.09248893, -0.12248101],
+            [ 0.04591711, -0.03486403, -0.07891743],
+            [ 0.93660195, -0.05762491,  0.20139844]])]
+
+    Beyond default models::
+
+        import scipy.special
+        import numpy as np
+        from sklearn.linear_model import LassoCV
+        from sklearn.ensemble import GradientBoostingClassifier, GradientBoostingRegressor
+        from econml.drlearner import DRLearner
+
+        np.random.seed(123)
+        X = np.random.normal(size=(1000, 3))
+        T = np.random.binomial(2, scipy.special.expit(X[:, 0]))
+        sigma = 0.01
+        y = (1 + .5*X[:, 0]) * T + X[:, 0] + np.random.normal(0, sigma, size=(1000,))
+        est = DRLearner(model_propensity=GradientBoostingClassifier(),
+                        model_regression=GradientBoostingRegressor(),
+                        model_final=LassoCV(cv=3),
+                        featurizer=None)
+        est.fit(y, T, X=X, W=None)
+
+    >>> est.score_
+    3.172135302455655
+    >>> est.const_marginal_effect(X[:3])
+    array([[ 0.55038338,  1.14558174],
+           [ 0.32065866,  0.75638221],
+           [-0.07514842, -0.03658315]])
+    >>> est.model_cate(T=2).coef_
+    array([ 0.86420672,  0.01628151, -0.        ])
+    >>> est.model_cate(T=2).intercept_
+    2.067552713536296
+    >>> est.model_cate(T=1).coef_
+    array([0.43487391, 0.02968939, 0.        ])
+    >>> est.model_cate(T=1).intercept_
+    0.9928852195090293
+
+    Attributes
+    ----------
+    models_propensity: list of objects of type(model_propensity)
+        A list of instances of the model_propensity object. Each element corresponds to a crossfitting
+        fold and is the model instance that was fitted for that training fold.
+    models_regression: list of objects of type(model_regression)
+        A list of instances of the model_regression object. Each element corresponds to a crossfitting
+        fold and is the model instance that was fitted for that training fold.
+    model_cate(T=t) : object of type(model_final)

```

Comment:
This is a method, not an attribute...  Why not put the docstring on the method itself?

Likewise, many of the other things are properties rather than attributes, as well.

Reply:
What I wanted to follow here is that properties that can be called only after fitted, I tried to make them attributes.

Reply:
I believe that in docs like https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html "Attribute" has a specific technical meaning: attributes are things that have been set directly on an instance's dictionary, like `self.coef_ = ...` (although admittedly this distinction is confusing, because the python spec and other sources describe all members as attributes, so really all properties are attributes, too).

Reply:
That makes sense. I'll revert to docstrings for these ones here, except score_.